### ISSUE TITLE
`sci-pt/picoscope-6.14.54-r4` does not start with the provided launcher script

### ISSUE THREAD

**Issue by bluecube** [AndrewAmmerlaan/natinst#2]
When I Install  and try to run the `/usr/bin/picoscope` script, it terminates without any output. Running it as 

`mono /opt/picoscope/lib/PicoScope.GTK.exe` I see the following error:

```
Unhandled Exception:
System.TypeLoadException: Could not load type of field 'Pico.:' (0) due to: Could not load file or assembly 'gtk-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f' or one of its dependencies.
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeLoadException: Could not load type of field 'Pico.:' (0) due to: Could not load file or assembly 'gtk-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f' or one of its dependencies.
```

I have tried installing `dev-dotnet/gtk-sharp-3.22.2-r1` from overlay, but the problem remains.
The solution to this is to also install `sci-pt/picomono` and run the program as ` /opt/picomono/bin/mono /opt/picoscope/lib/PicoScope.GTK.exe`

Then there is an additional error with libpng version, Ive added it into a separate issue (#1).

I think the proper solution would be to add the `picomono` dependency and a new launcher script that uses it.

I tried to do this as a PR, but kind of got stuck on your ebuild generator scripts :-). Anyway, thanks a lot for providing this overlay it is very helpful.

**Comment by Nowa-Ammerlaan** [comment]
To be honest, I provide these packages in the overlay because with the scripts it's almost no extra effort to generate them. However, I never actually managed to get it working. If I run it with picomono I get this weird error. Possibly because I am missing gtk-sharp, which overlay did you install it from? Or is gtk-sharp also provided by picomono?

```
andrew@andrew-gentoo-pc / % /opt/picomono/bin/mono /opt/picoscope/lib/PicoScope.GTK.exe
Gtk-Message: 09:07:16.220: Failed to load module "canberra-gtk-module"

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.Exception: Magic number is wrong: 542
at System.TermInfoReader.ReadHeader (System.Byte[] buffer, System.Int32& position) [0x0002b] in <8f2c484307284b51944a1a13a14c0266>:0 
at System.TermInfoReader..ctor (System.String term, System.String filename) [0x00065] in <8f2c484307284b51944a1a13a14c0266>:0 
at System.TermInfoDriver..ctor (System.String term) [0x00058] in <8f2c484307284b51944a1a13a14c0266>:0 
at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0 
at System.ConsoleDriver..cctor () [0x00062] in <8f2c484307284b51944a1a13a14c0266>:0 
--- End of inner exception stack trace ---
at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x0000a] in <8f2c484307284b51944a1a13a14c0266>:0
at System.Console..cctor () [0x000a8] in <8f2c484307284b51944a1a13a14c0266>:0
--- End of inner exception stack trace ---
at System.Diagnostics.ConsoleTraceListener..ctor () [0x00000] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0
at (wrapper remoting-invoke-with-check) System.Diagnostics.ConsoleTraceListener:.ctor ()
at Pico.. () [0x0003b] in <5739de285ebd49f9b734bea8ca65242b>:0
at Pico..Main () [0x00040] in <5739de285ebd49f9b734bea8ca65242b>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.Exception: Magic number is wrong: 542
at System.TermInfoReader.ReadHeader (System.Byte[] buffer, System.Int32& position) [0x0002b] in <8f2c484307284b51944a1a13a14c0266>:0
at System.TermInfoReader..ctor (System.String term, System.String filename) [0x00065] in <8f2c484307284b51944a1a13a14c0266>:0
at System.TermInfoDriver..ctor (System.String term) [0x00058] in <8f2c484307284b51944a1a13a14c0266>:0
at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0
at System.ConsoleDriver..cctor () [0x00062] in <8f2c484307284b51944a1a13a14c0266>:0
--- End of inner exception stack trace ---
at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x0000a] in <8f2c484307284b51944a1a13a14c0266>:0
at System.Console..cctor () [0x000a8] in <8f2c484307284b51944a1a13a14c0266>:0
--- End of inner exception stack trace ---
at System.Diagnostics.ConsoleTraceListener..ctor () [0x00000] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0
at (wrapper remoting-invoke-with-check) System.Diagnostics.ConsoleTraceListener:.ctor ()
at Pico.. () [0x0003b] in <5739de285ebd49f9b734bea8ca65242b>:0
at Pico..Main () [0x00040] in <5739de285ebd49f9b734bea8ca65242b>:0
```

> I tried to do this as a PR, but kind of got stuck on your ebuild generator scripts :-)

I think we'll just have to do this manually directly in the ebuild. Putting this in the script would require making a special case of picoscope. Besides, the package updates rather infrequently so it is not a lot of extra work to do some small manual steps on each regeneration.

**Comment by Nowa-Ammerlaan** [comment]
> To be honest, I provide these packages in the overlay because with the scripts it's almost no extra effort to generate them. However, I never actually managed to get it working. If I run it with picomono I get this weird error. Possibly because I am missing gtk-sharp, which overlay did you install it from? Or is gtk-sharp also provided by picomono?
> 
> ```
> andrew@andrew-gentoo-pc / % /opt/picomono/bin/mono /opt/picoscope/lib/PicoScope.GTK.exe
> Gtk-Message: 09:07:16.220: Failed to load module "canberra-gtk-module"
> 
> Unhandled Exception:
> System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.Exception: Magic number is wrong: 542
> at System.TermInfoReader.ReadHeader (System.Byte[] buffer, System.Int32& position) [0x0002b] in <8f2c484307284b51944a1a13a14c0266>:0 
> at System.TermInfoReader..ctor (System.String term, System.String filename) [0x00065] in <8f2c484307284b51944a1a13a14c0266>:0 
> at System.TermInfoDriver..ctor (System.String term) [0x00058] in <8f2c484307284b51944a1a13a14c0266>:0 
> at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0 
> at System.ConsoleDriver..cctor () [0x00062] in <8f2c484307284b51944a1a13a14c0266>:0 
> --- End of inner exception stack trace ---
> at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x0000a] in <8f2c484307284b51944a1a13a14c0266>:0
> at System.Console..cctor () [0x000a8] in <8f2c484307284b51944a1a13a14c0266>:0
> --- End of inner exception stack trace ---
> at System.Diagnostics.ConsoleTraceListener..ctor () [0x00000] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0
> at (wrapper remoting-invoke-with-check) System.Diagnostics.ConsoleTraceListener:.ctor ()
> at Pico.. () [0x0003b] in <5739de285ebd49f9b734bea8ca65242b>:0
> at Pico..Main () [0x00040] in <5739de285ebd49f9b734bea8ca65242b>:0
> [ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.Exception: Magic number is wrong: 542
> at System.TermInfoReader.ReadHeader (System.Byte[] buffer, System.Int32& position) [0x0002b] in <8f2c484307284b51944a1a13a14c0266>:0
> at System.TermInfoReader..ctor (System.String term, System.String filename) [0x00065] in <8f2c484307284b51944a1a13a14c0266>:0
> at System.TermInfoDriver..ctor (System.String term) [0x00058] in <8f2c484307284b51944a1a13a14c0266>:0
> at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0
> at System.ConsoleDriver..cctor () [0x00062] in <8f2c484307284b51944a1a13a14c0266>:0
> --- End of inner exception stack trace ---
> at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x0000a] in <8f2c484307284b51944a1a13a14c0266>:0
> at System.Console..cctor () [0x000a8] in <8f2c484307284b51944a1a13a14c0266>:0
> --- End of inner exception stack trace ---
> at System.Diagnostics.ConsoleTraceListener..ctor () [0x00000] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0
> at (wrapper remoting-invoke-with-check) System.Diagnostics.ConsoleTraceListener:.ctor ()
> at Pico.. () [0x0003b] in <5739de285ebd49f9b734bea8ca65242b>:0
> at Pico..Main () [0x00040] in <5739de285ebd49f9b734bea8ca65242b>:0
> ```
> 
> > I tried to do this as a PR, but kind of got stuck on your ebuild generator scripts :-)
> 
> I think we'll just have to do this manually directly in the ebuild. Putting this in the script would require making a special case of picoscope. Besides, the package updates rather infrequently so it is not a lot of extra work to do some small manual steps on each regeneration.

Okay so apparently this is also required `export TERM=xterm`. Now it works! I'll see if I can get this stuff into the ebuild.

**Comment by Nowa-Ammerlaan** [comment]
> I think the proper solution would be to add the picomono dependency and a new launcher script that uses it.

I patched the launch script to hardcode `MONO_CMD` to `/opt/picomono/bin/mono`. Another way to do it would be to add `/opt/picomono/bin` to `PATH` (then the call to `which` should succeed), but then we may run into strange conflicts with `dev-lang/mono` so I opted for your solution instead.

`dev-pt/*` is basically untested, so let me know if there are any further issues.
